### PR TITLE
Introduce obfuscate module with tests

### PR DESCRIPTION
**Pull request by gauntface**
I hit an issue when running this locally and narrowed it down to the obfuscate method.

One additional layer here is to add a rescue exception clause which it looks like is being done in production, but it's unclear if that's necessary with this approach, but given the nature of this tool, it might be the best option to keep on going.

### CODE REVIEW

File: crawler/crawler.rb [hunk 1]
```diff
@@ -90,11 +82,12 @@
           @log.info "Missed records.."
         end
 
-        StatHat.new.ez_count('Github Events', new_events.size)
+        # StatHat.new.ez_count('Github Events', new_events.size)
```
Comment (igrigorik) [inline crawler/crawler.rb:85]: You want to keep this around for stats.

File: crawler/crawler.rb [hunk 2]
```diff
@@ -90,11 +82,12 @@
...
 
       rescue Exception => e
         @log.error "Processing exception: #{e}, #{e.backtrace.first(5)}"
         @log.error "Response: #{req.response_header}, #{req.response}"
+        raise "Stop"
```
Comment (igrigorik) [inline crawler/crawler.rb:90]: I don't think you want to re-raise here. We don't want to abort the process if we hit an error on a batch: log it, continue.

File: crawler/obfuscate.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        if h.is_a? Hash
+            if h['email'] && !h['email'].empty?
+                email = h.delete('email')
+                name, host = email.split("@")
+                h['email'] = [Digest::SHA1.hexdigest(name), host].compact.join("@")
+            end
+
+            h.each_value do |v|
+                self.email(v) if v.is_a? Hash
+                v.each {|e| self.email(e)} if v.is_a? Array
```
Comment (igrigorik) [inline crawler/obfuscate.rb:12]: There is an unhandled edge case here that recently took down the crawler. I believe the issue was that we had an array of strings and that crashed this function.

**Review by igrigorik — COMMENTED** [review]
👍

**Comment by gauntface** [comment]
Sorry I think I gave you a bunch of debug code in this PR. Will tidy up ASAP.

File: crawler/crawler.rb (hunk 2 above)
Comment (gauntface) [reply crawler/crawler.rb:90]: Sorry this was debug code.

File: crawler/crawler.rb (hunk 1 above)
Comment (gauntface) [reply crawler/crawler.rb:85]: Sorry this was debug code.

File: crawler/obfuscate.rb (hunk 3 above)
Comment (gauntface) [reply crawler/obfuscate.rb:12]: Yeah I believe this is captured in the test - see test_no_obfuscation_for_other_fields which has an "array-string" value.

This bubbled up in local testing and was before I found the fix in prod.

**Review by igrigorik — APPROVED** [review]
👍🏻 nice work!

**Pull request merged**